robot-name/robot_name.go

```diff
@@ -0,0 +1,83 @@
+// Manage robot factory settings.
+package robotname
+
+import (
+	"errors"
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+// Robot has a name
+type Robot struct {
+	name string
+}
+
+// All the existing robot names so we can check for unique names
+var robots = make(map[string]Robot)
```

Comment:
You never use the value for this map, so you could declare it as `map[string]struct{}`. Adding a value to this map would look like `robots[key] = struct{}{}`. Good thing about this is that it doesn't use extra memory to hold the value. Empty struct don't use space.